core/src/main/resources/reference.conf

```diff
@@ -225,6 +225,15 @@ akka.persistence.cassandra {
     # replays.
     deserialization-parallelism = 1
 
+    # For applications that are not using persistenceIds or currentPersistenceIds queries
+    # this can be set to 'off', which will optimize the write of the first event for each
+    # persistent actor since the all_persistence_id table doesn't have to be populated.
+    # Note that the Cleanup and Reconcilation tools may also use persistenceIds queries
+    # and those will not work if this was disabled when the events were written. In
+    # that case the all_persistence_id table can be reconstructed with
+    # Reconcilation.rebuildAllPersistenceIds.
+    support-all-persistence-ids = on
```

Comment:
not sure this is the best property name, and if it should be in the `query` or `journal` section?
we have similar `journal.support-deletes` and `events-by-tag.enabled`.

Reply:
I think journal as it is enabling the feature. We could make the query plugin fall back to the old inefficient query

Reply:
ok, moving it to the `journal`. I don't think we should silently fall back to the inefficient query. Better to know about it, and run the migration job once.